app/lib/logic/cms.rb

```diff
@@ -37,7 +37,7 @@ module Provider
           cms.has_many :all_partials,         class_name: 'CMS::Partial'
           cms.has_many :portlets,             class_name: 'CMS::Portlet'
 
-          cms.has_many :provided_groups,      class_name: 'CMS::Group', dependent: :delete_all
+          cms.has_many :provided_groups,      class_name: 'CMS::Group', dependent: :destroy
```

Comment:
Without this change, there is nothing to delete `group_sections` from database as it is only referred through group or section. So group or section needs to use `destroy`. I found `group` is simpler so less overhead. Additionally we don't seem to have a section_id index in `group_sections` table so `group` must be it.